src/sql/workbench/contrib/notebook/browser/notebookActions.ts

```diff
@@ -272,6 +271,102 @@ export class CollapseCellsAction extends ToggleableAction {
+	private static readonly iconClass = 'icon-run-with-parameters';
+
+	constructor(id: string,
+		toggleTooltip: boolean,
+		context: URI,
+		@IQuickInputService private quickInputService: IQuickInputService,
+		@INotebookService private _notebookService: INotebookService,
+		@INotificationService private notificationService: INotificationService,
+	) {
+		super(id, {
+			label: RunParametersAction.label,
+			baseClass: baseIconClass,
+			iconClass: RunParametersAction.iconClass,
+			maskedIconClass: maskedIconClass,
+			shouldToggleTooltip: toggleTooltip
+		});
+	}
+
+	/**
+	 * Gets Default Parameters in Notebook from Parameter Cell
+	 * Uses that as Placeholder values for user to inject new values for
+	 * Once user enters all values it will open the new parameterized notebook
+	 * with injected parameters value from the QuickInput
+	*/
+	public async run(context: URI): Promise<void> {
+		const editor = this._notebookService.findNotebookEditor(context);
+		// Set defaultParameters to the parameter values in parameter cell
+		let defaultParameters = new Map<string, string>();
+		editor.cells.forEach(cell => {
+			if (cell.isParameter) {
+				for (let parameter of cell.source) {
+					let param = parameter.split('=', 2);
+					defaultParameters.set(param[0].trim(), param[1].trim());
+				}
+			}
+		});
+
+		// Store new parameters values the user inputs
+		let inputParameters = new Map<string, string>();
+		let uriParams = new URLSearchParams();
+		// Store new parameter values to map based off defaultParameters
+		if (defaultParameters.size === 0) {
+			// If there is no parameter cell indicate to user to create one
+			this.notificationService.notify({
+				severity: Severity.Info,
+				message: localize('noParametersCell', "This notebook cannot run with parameters until a parameter cell is added. [Learn more](https://docs.microsoft.com/sql/azure-data-studio/notebooks/notebooks-parameterization)."),
+			});
+			return;
+		} else {
+			for (let key of defaultParameters.keys()) {
+				let newParameterValue = await this.quickInputService.input({ prompt: key, value: defaultParameters.get(key) });
+				// If user cancels or escapes then it stops the action entirely
+				if (newParameterValue === undefined) {
+					return;
+				}
+				inputParameters.set(key, newParameterValue);
+			}
+			// Format the new parameters to be append to the URI
+			for (let key of inputParameters.keys()) {
+				// Will only add new injected parameters when the value is not the same as the defaultParameters values
+				if (inputParameters.get(key) !== defaultParameters.get(key)) {
+					// For empty strings we need to escape the value
+					// so that it is kept when adding uriParams.toString() to filePath
+					if (inputParameters.get(key) === '') {
+						uriParams.append(key, '\'\'');
+					} else {
+						uriParams.append(key, inputParameters.get(key));
+					}
+				}
+			}
+			let filePath = context.path;
+			filePath = filePath + '?' + unescape(uriParams.toString());
```

Comment:
Why are you unescaping this?

1.  I would expect `openParameterizedNotebook` to take in a URI instead of just a string
2. Something else should be doing the unescaping, whenever you have something in URI form I would expect that it's escaped properly (and only unescape it when you extra out the query in the place that's going to use it)

Reply:
I looked at this further - why are you even taking in both the URI and the filepath anyways? They both should be coming from the URI I would imagine

Reply:
So `openParameterizedNotebook` follows the same logic to how notebookURIHandler is setup for URI Parameterization. 

We must take the filePath as a string with the correct unescaped parameters in order for when we build the new uri from this filePath it will story the query parameters correctly in the uri.query. 
This would then be consumed properly on the notebookModel side as that is what we are able to retrieve from the uri. 
I need to unescape the uriParams.toString() as we discussed that this would be not correctly formed to how we would expect it - `Input.ipynb?x=1&y=2'`. 

The uri that is being passed in is of the currentNotebook which will get us the modelContents. The filePath is of the current notebook with the new parameters added to it which would then be parsed to create a new URI.

Reply:
The other way to solely pass the URI would be to utilize the current URI and then set the uriParams query to the query. 
`x= unescape(uriParams.toString())
uri.with({query: x})` before passing it to `openParameterizedNotebook`.

Reply:
Yup, that's exactly what I'm suggesting.

AFAIK we're kind of in unprecedented territory here - I don't think any of the current editors support extra parameters like this. But doing it this way makes sense to me since we already pass in a URI and this would keep things consistent with how the file URIs are passed around